src/ScopedFd.h

```diff
@@ -17,7 +19,6 @@ namespace rr {
 class ScopedFd {
 public:
   ScopedFd() : fd(-1) {}
-  ScopedFd(int fd) : fd(fd) {}
```

Comment:
```c++
    explicit ScopedFd(int fd) : fd(fd) {}
```
Keno wanted me to mention that this constructor could also be kept, with the `explicit` keyword to prevent it from being quite as easily called unintentionally. It still has a few more cases, IIRC, where this could get used unintentionally, but it cuts down on the worse of the issues.

Reply:
@rocallahan can say which he prefers. I think either is fine. Using `explicit` is certainly a smaller change. (I would have used it, but I failed to remember what it was called).

Reply:
I think making the constructor `explicit` is the right approach here. We probably should avoid non-explicit 1-arg constructors everywhere. It's a design error in C++ that single-arg constructors are used as implicit conversions by default.